### ISSUE TITLE
Performance of the model

### ISSUE THREAD

**Issue by DriraYosr** [clovaai/donut#304]
I don't know how to choose the best number of epochs. Is there a way to evaluate the model's performance throughout  the epochs used in training ?

**Comment by kirby707** [comment]
The best number of epochs actually depends on your dataset. For example, the best number of epochs will be different for a small (and not-so-diverse) dataset vs. a large (and diverse) dataset.

A common practice is to split your dataset into Train and Validation splits and visualize Training Loss and Validation Loss for each epoch. The Training Loss usually goes down epoch after epoch but the Validation Loss usually reaches a minimum point and then starts going up again. Sometimes, the Validation Loss reaches the minimum point and just stays the same for the rest of the training run.

In either case, the epoch where the Validation Loss was the lowest: technically that is where the model showed the best performance (that model checkpoint should be able to analyze unseen data better than any other checkpoint).
